go/vt/sqlparser/sql.y

```diff
@@ -1124,7 +1126,7 @@ function_call_keyword:
   {
     $$ = &ConvertExpr{Expr: $3, Type: $5}
   }
-| MATCH openb column_list closeb AGAINST openb value_expression match_option closeb
+| MATCH openb column_name_list closeb AGAINST openb value_expression match_option closeb
```

Comment:
I think select_expression_list could just work here. Then we can avoid the new type. Can you try and let me know?